python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -90,37 +114,42 @@ def cuda_array_interface_wrapper(
     )
 
 
-class Buffer(Serializable):
-    """A Buffer represents device memory.
+class BufferOwner(Serializable):
+    """A owning buffer that represents device memory.
 
-    Use the factory function `as_buffer` to create a Buffer instance.
+    This class isn't meant to be used throughout cuDF. Instead, it
+    standardizes data owning by wrapping any data object that
+    represents device memory. Multiple `Buffer` instances, which are
+    the ones used throughout cuDF, can then refer to the same
+    `BufferOwner` instance.
+
+    Use `_from_device_memory` and `_from_host_memory` to create
+    a new instance from either device or host memory respectively.
     """
 
     _ptr: int
     _size: int
     _owner: object
 
-    def __init__(self):
-        raise ValueError(
-            f"do not create a {self.__class__} directly, please "
-            "use the factory function `cudf.core.buffer.as_buffer`"
-        )
-
     @classmethod
-    def _from_device_memory(cls, data: Any) -> Self:
-        """Create a Buffer from an object exposing `__cuda_array_interface__`.
+    def _from_device_memory(cls, data: Any, exposed: bool) -> Self:
+        """Create from an object exposing `__cuda_array_interface__`.
 
         No data is being copied.
 
         Parameters
         ----------
         data : device-buffer-like
             An object implementing the CUDA Array Interface.
+        exposed : bool
+            Mark the buffer as permanently exposed. This is used by
+            ExposureTrackedBuffer to determine when a deep copy is required
+            and by SpillableBuffer to mark the buffer unspillable.
```

Comment:
So this is ignored in this class and only makes sense in the exposuretrackedbuffer object, right?

Reply:
yes

Reply:
`ExposureTrackedBufferOwner` has been merged with `BufferOwner`